CRM/Mailing/Event/BAO/Opened.php

```diff
@@ -241,11 +241,24 @@ public static function &getRows(
     $contact = CRM_Contact_BAO_Contact::getTableName();
     $email = CRM_Core_BAO_Email::getTableName();
 
+    $selectClauses = array(
+      "$contact.display_name as display_name",
+      "$contact.id as contact_id",
+      "$email.email as email",
+      "{$open}.time_stamp as date",
+    );
+
+    if ($is_distinct) {
+      $groupBy = " GROUP BY $queue.id ";
+      $select = CRM_Contact_BAO_Query::appendAnyValueToSelect($selectClauses, "{$open}.time_stamp as date");
```

Comment:
I agree that the existing group by is incorrect. 

It seems the effect of this is to avoid grouping by these other fields. In the case of display_name, contact_id and email they will always be the same so perhaps there is a performance gain in omitting them from the group by?

Regarding the time_stamp. Depending on the platform mysql is installed on this will either retrieve the earliest timestamp (equivalent to making it min(time_stamp)) OR any time stamp - is this desirable / fine? In particular the latter variant?